build/build.go

```diff
@@ -929,6 +933,13 @@ func BuildWithResultHandler(ctx context.Context, nodes []builder.Node, opt map[s
 								printRes = res.Metadata
 							}
 							results.Set(resultKey(dp.driverIndex, k), res)
+							if node.Driver.Features()[driver.HistoryAPI] && desktop.BuildBackendEnabled() {
+								buildDetailsOpts[k] = desktop.BuildDetailsOpt{
+									Builder: node.Builder,
+									Node:    node.Name,
+									Ref:     so.Ref,
+								}
+							}
 							if resultHandleFunc != nil {
 								resultCtx, err := NewResultContext(ctx, cc, so, res)
```

Comment:
@jedevc If we want to print in commands pkg I was wondering if we could use the result context handler to set the build details opts?

Reply:
`ResultContext` doesn't leave outside the controller boundary, so that would likely only work for the `runBasicBuild` case I think :cry: Although, we do record the build options, so `controller.Inspect` could work for this?

Though that's only the controller options, not the full solve request.

Reply:
Ah right thanks for the feedback, I will take a closer look.

Reply:
Builder name, node name when driver pairs are computed during build are not inputs so doesn't look good to have it in build options. Same for the `refID` computed for each solve opt.

Reply:
Opened https://github.com/docker/buildx/pull/1861 to carry changes from #1680. Will revisit changes here once switching bake to controller for building.